src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs

```diff
@@ -23,7 +23,8 @@ internal SslAuthenticationOptions(SslClientAuthenticationOptions sslClientAuthen
             EncryptionPolicy = sslClientAuthenticationOptions.EncryptionPolicy;
             IsServer = false;
             RemoteCertRequired = true;
-            TargetHost = sslClientAuthenticationOptions.TargetHost!;
+            // RFC 6066 section 3 says to exclude trailing dot from fully qualified DNS hostname
+            TargetHost = sslClientAuthenticationOptions.TargetHost!.TrimEnd('.');
```

Comment:
This might remove multiple trailing dots.  Is that correct?

Reply:
multiple subsequent dots are not valid in name.(and trailing dot is extremely rare).  DNS lookup would likely fail but I'm not sure if we care here. It feels there would be no functional harm.

Reply:
is the `!` necessary here? I believe the assert above should suppress it.

Reply:
good point, removed